apps/arena/lib/arena/matchmaking/deathmatch_mode.ex

```diff
@@ -60,19 +58,43 @@ defmodule Arena.Matchmaking.DeathmatchMode do
     Process.send_after(self(), :launch_game?, 300)
     diff = System.monotonic_time(:millisecond) - state.batch_start_at
 
-    if length(clients) >= Application.get_env(:arena, :players_needed_in_match) or
-         (diff >= Utils.start_timeout_ms() and length(clients) > 0) do
+    state =
+      if Map.has_key?(state, :game_mode_configuration) do
+        state
+      else
+        case Arena.Configuration.get_game_mode_configuration("deathmatch", "deathmatch") do
+          {:error, _} ->
+            state
+
+          {:ok, game_mode_configuration} ->
+            # This is needed because we might not want to send a request every 300 seconds to the game backend
+            Process.send_after(self(), :update_params, 5000)
+
+            map = Enum.random(game_mode_configuration.map_mode_params)
+
+            Map.put(state, :game_mode_configuration, game_mode_configuration)
+            |> Map.put(:current_map, map)
+        end
+      end
+
+    if Map.has_key?(state, :game_mode_configuration) &&
+         (length(clients) >= state.current_map.amount_of_players or
+            (diff >= Utils.start_timeout_ms() and length(clients) > 0)) do
       send(self(), :start_game)
     end
 
     {:noreply, state}
   end
 
   def handle_info(:start_game, state) do
-    {game_clients, remaining_clients} = Enum.split(state.clients, Application.get_env(:arena, :players_needed_in_match))
-    create_game_for_clients(game_clients)
+    {game_clients, remaining_clients} =
+      Enum.split(state.clients, state.current_map.amount_of_players)
+
+    create_game_for_clients(game_clients, state.game_mode_configuration, state.current_map)
 
-    {:noreply, %{state | clients: remaining_clients}}
+    map = Enum.random(state.game_mode_configuration.map_mode_params)
```

Comment:
Why are we doing this? Couldn't we fall in an error where we use a different current_map to create_game_for_clients and then override a different one in the state.current_map?

Reply:
done in [chore: send the update params event in the init of each matchmaking](https://github.com/lambdaclass/mirra_backend/pull/1061/commits/b3df822b0785c0bb230428488125c6915a79964d)